client/modules/IDE/components/ConsoleInput.jsx

```diff
@@ -47,92 +41,84 @@ class ConsoleInput extends React.Component {
             messages
           }
         });
-        this.props.dispatchConsoleEvent(consoleEvent);
+        dispatchConsoleEvent(consoleEvent);
         cm.setValue('');
-        this.setState((state) => ({
-          commandCursor: -1,
-          commandHistory: [value, ...state.commandHistory]
-        }));
+        setCommandCursor(-1);
+        setCommandHistory((prevHistory) => [value, ...prevHistory]);
       } else if (e.key === 'ArrowUp') {
-        const lineNumber = this._cm.getDoc().getCursor().line;
+        const lineNumber = cm.getDoc().getCursor().line;
         if (lineNumber !== 0) {
           return false;
         }
 
-        this.setState((state) => {
-          const newCursor = Math.min(
-            state.commandCursor + 1,
-            state.commandHistory.length - 1
-          );
-          this._cm.getDoc().setValue(state.commandHistory[newCursor] || '');
-          const cursorPos = this._cm.getDoc().getLine(0).length - 1;
-          this._cm.getDoc().setCursor({ line: 0, ch: cursorPos });
-          return { commandCursor: newCursor };
+        setCommandCursor((prevCursor) => {
+          const newCursor = Math.min(prevCursor + 1, commandHistory.length - 1);
+          cm.getDoc().setValue(commandHistory[newCursor] || '');
```

Comment:
It seems not quite right to be performing these side effects inside of a set state.  But that's how it was before so let's leave it for now.